core/web/api/analytics.py

```diff
@@ -78,13 +85,19 @@ class OneShotAnalytics(CrudApi):
     def index(self):
         data = []
 
+        yeti_user = False
+        if current_user.username != 'yeti':
+            try:
+                yeti_user = User.objects.get(username='yeti')
+            except Exception as e:
+                logging.info(e)
+
         for obj in self.objectmanager.objects.all():
             info = obj.info()
 
             info['available'] = True
-            if hasattr(
-                    obj,
-                    'settings') and not current_user.has_settings(obj.settings):
+            if hasattr(obj, 'settings') and (not current_user.has_settings(obj.settings) and \
```

Comment:
please split into boolean variables before evaluation, this is very hard to read